api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java

```diff
@@ -295,4 +296,34 @@ public double getUserFreeTierDollarLimit(DbUser user) {
     return Optional.ofNullable(user.getFreeTierCreditsLimitDollarsOverride())
         .orElse(workbenchConfigProvider.get().billing.defaultFreeCreditsDollarLimit);
   }
+
+  /**
+   * Set a custom Free Tier dollar limit override for this user. If this is greater than the user's
+   * total cost, set their workspaces to active. Note: lowering the limit below total cost will not
+   * set the workspaces to inactive. checkFreeTierBillingUsage() will do this as part of the next
+   * cron run.
+   *
+   * @param user the user as represented in our database
+   * @param dollarLimit the US dollar amount, represented as a double
+   * @return whether the user's total cost is now below their limit
+   */
+  public boolean setFreeTierDollarOverride(DbUser user, double dollarLimit) {
+    final Double previousLimitMaybe = user.getFreeTierCreditsLimitDollarsOverride();
+
+    // TODO: prevent setting this limit directly except in this method?
+    user.setFreeTierCreditsLimitDollarsOverride(dollarLimit);
+    user = userDao.save(user);
```

Comment:
I am surprised to see this being set in this service as opposed to the UserAdminService or ProfileService or someplace. I guess you'd have to invert the dependency for that to happen though.